crates/datasources/src/common/query.rs

```diff
@@ -0,0 +1,116 @@
+use std::fmt::Write;
+
+use datafusion::arrow::datatypes::SchemaRef;
+use datafusion::error::DataFusionError;
+use datafusion::logical_expr::Expr;
+
+use super::errors::Result;
+use super::util::{self, Datasource};
+
+pub fn from_exprs(
```

Comment:
we shouldn't use the module to provide context for a function name. 

`from_exprs` has no context without the `query::` part. `query::from_exprs`

```suggestion
pub fn exprs_to_query_string(
```

Reply:
I'd also proceed with caution as not all sql dialects are the same, as we add more expr pushdowns, we'll need to specialize for different dialects.

Reply:
> I'd also proceed with caution as not all sql dialects are the same, as we add more expr pushdowns, we'll need to specialize for different dialects.

the code in question already has specialization for this. 

> we shouldn't use the module to provide context for a function name.

meh, I disagree

- https://github.com/rust-lang/rfcs/blob/master/text/0356-no-module-prefixes.md
- https://github.com/rust-lang/api-guidelines/discussions/66

anyway, this isn't the right place for this conversation.

Reply:
i think this is slightly different than the examples you provided. 

an error inside `io::Error` is still an error, but more specifically it is an IO error, so any of `IOError`, `io::Error`, `Error` still indicate that it's an error. `io` just provides additional context. same with the `log::LogLevel` example. 

`from_exprs` is completely meaningless without `query::from_exprs`.